### PR TITLE
add some semantic function of instructions

### PR DESCRIPTION
**Pull request by yjxxin**
When I added semantic functions, I found that there are many instructions in remill that did not add corresponding semantic functions, such as VPMOVSXBQ and VPMOVSXWD. I added more than 300 similar instructions when doing related work, such as

```
namespace {
template <typename D, typename K, typename S>
DEF_SEM(VPMOVSXBQ_MASKmskw_SIMD128, D dst, K k1, S src) {
  auto src_vec = SReadV8(src);
  auto dst_vec = SClearV64(SReadV64(dst));
  auto k_vec = Read(k1);
  for (int i = 0; i< 2; i++) {
    if (READBIT(k_vec, i) == 0) {
      dst_vec = SInsertV64(dst_vec, i, 0);
    }else {
      auto v = SExtTo<int64_t>(SExtractV8(src_vec, i));
      dst_vec = SInsertV64(dst_vec, i, v);
    }
  }
  SWriteV64(dst, dst_vec);
  return memory;
}
}

#if HAS_FEATURE_AVX512
DEF_ISEL(VPMOVSXBQ_XMMi64_MASKmskw_MEMi8_AVX512) = VPMOVSXBQ_MASKmskw_SIMD128<VV128W, R8, MV16>;
DEF_ISEL(VPMOVSXBQ_XMMi64_MASKmskw_XMMi8_AVX512) = VPMOVSXBQ_MASKmskw_SIMD128<VV128W, R8, V128>;
#endif  // HAS_FEATURE_AVX512

namespace {
template <typename D, typename K, typename S>
DEF_SEM(VPMOVSXWD_MASKmskw_SIMD128, D dst, K k1, S src) {
  auto src_vec = SReadV16(src);
  auto dst_vec = SClearV32(SReadV32(dst));
  auto k_vec = Read(k1);
  for (int i = 0; i< 4; i++) {
    if (READBIT(k_vec, i) == 0) {
      dst_vec = SInsertV32(dst_vec, i, 0);
    }else {
      auto v = SExtTo<int32_t>(SExtractV16(src_vec, i));
      dst_vec = SInsertV32(dst_vec, i, v);
    }
  }
  SWriteV32(dst, dst_vec);
  return memory;
}
}

#if HAS_FEATURE_AVX512
DEF_ISEL(VPMOVSXWD_XMMi32_MASKmskw_MEMi16_AVX512) = VPMOVSXWD_MASKmskw_SIMD128<VV128W, R8, MV64>;
DEF_ISEL(VPMOVSXWD_XMMi32_MASKmskw_XMMi16_AVX512) = VPMOVSXWD_MASKmskw_SIMD128<VV128W, R8, V128>;
#endif  // HAS_FEATURE_AVX512
```
If you are satisfied, I can upload all the instructions I have added to the community to help remill expand and support conversion of more instructions.

### CODE REVIEW

File: lib/Arch/X86/Semantics/DATAXFER.cpp [hunk 1]
```diff
@@ -1121,3 +1121,51 @@ IF_64BIT(DEF_ISEL(MOVSXD_GPRv_MEMd_64) = MOVSX<R64W, M32, int64_t>;)
...
+namespace {
+template <typename D, typename K, typename S>
+DEF_SEM(VPMOVSXBQ_MASKmskw_SIMD128, D dst, K k1, S src) {
+  auto src_vec = SReadV8(src);
+  auto dst_vec = SClearV64(SReadV64(dst));
+  auto k_vec = Read(k1);
+  for (int i = 0; i< 2; i++) {
+    if (READBIT(k_vec, i) == 0) {
+      dst_vec = SInsertV64(dst_vec, i, 0);
+    }else {
```
Comment (pgoodman) [inline lib/Arch/X86/Semantics/DATAXFER.cpp:1134]: `} else {`

File: lib/Arch/X86/Semantics/DATAXFER.cpp [hunk 2]
```diff
@@ -1121,3 +1121,51 @@ IF_64BIT(DEF_ISEL(MOVSXD_GPRv_MEMd_64) = MOVSX<R64W, M32, int64_t>;)
...
 IF_64BIT(DEF_ISEL(MOVSXD_GPRv_GPR32_64) = MOVSX<R64W, R32, int64_t>;)
 IF_64BIT(DEF_ISEL(MOVSXD_GPRv_GPRz_64) = MOVSX<R64W, R32, int64_t>;)
+
+namespace {
+template <typename D, typename K, typename S>
+DEF_SEM(VPMOVSXBQ_MASKmskw_SIMD128, D dst, K k1, S src) {
+  auto src_vec = SReadV8(src);
+  auto dst_vec = SClearV64(SReadV64(dst));
+  auto k_vec = Read(k1);
+  for (int i = 0; i< 2; i++) {
```
Comment (pgoodman) [inline lib/Arch/X86/Semantics/DATAXFER.cpp:1131]: `i < 2`

Comment (pgoodman) [reply lib/Arch/X86/Semantics/DATAXFER.cpp:1131]: `auto i = 0u`

File: lib/Arch/X86/Semantics/DATAXFER.cpp [hunk 3]
```diff
@@ -1121,3 +1121,51 @@ IF_64BIT(DEF_ISEL(MOVSXD_GPRv_MEMd_64) = MOVSX<R64W, M32, int64_t>;)
...
+      auto v = SExtTo<int64_t>(SExtractV8(src_vec, i));
+      dst_vec = SInsertV64(dst_vec, i, v);
+    }
+  }
+  SWriteV64(dst, dst_vec);
+  return memory;
+}
+}
+
+#if HAS_FEATURE_AVX512
```
Comment (pgoodman) [inline lib/Arch/X86/Semantics/DATAXFER.cpp:1144]: Move this `#if` up to encompass the `VPMOVSXBQ_MASKmskw_SIMD128` semantic.

File: lib/Arch/X86/Semantics/DATAXFER.cpp [hunk 4]
```diff
@@ -1121,3 +1121,51 @@ IF_64BIT(DEF_ISEL(MOVSXD_GPRv_MEMd_64) = MOVSX<R64W, M32, int64_t>;)
...
+DEF_ISEL(VPMOVSXBQ_XMMi64_MASKmskw_MEMi8_AVX512) = VPMOVSXBQ_MASKmskw_SIMD128<VV128W, R8, MV16>;
+DEF_ISEL(VPMOVSXBQ_XMMi64_MASKmskw_XMMi8_AVX512) = VPMOVSXBQ_MASKmskw_SIMD128<VV128W, R8, V128>;
+#endif  // HAS_FEATURE_AVX512
+
+namespace {
+template <typename D, typename K, typename S>
+DEF_SEM(VPMOVSXWD_MASKmskw_SIMD128, D dst, K k1, S src) {
```
Comment (pgoodman) [inline lib/Arch/X86/Semantics/DATAXFER.cpp:1151]: Probably, you could go and group these avx512-related semantics together, then follow them up with the isel defs, all within the `#if` blocks.

File: lib/Arch/X86/Semantics/DATAXFER.cpp [hunk 5]
```diff
@@ -1121,3 +1121,51 @@ IF_64BIT(DEF_ISEL(MOVSXD_GPRv_MEMd_64) = MOVSX<R64W, M32, int64_t>;)
...
+  auto src_vec = SReadV16(src);
+  auto dst_vec = SClearV32(SReadV32(dst));
+  auto k_vec = Read(k1);
+  for (int i = 0; i< 4; i++) {
```
Comment (pgoodman) [inline lib/Arch/X86/Semantics/DATAXFER.cpp:1155]: `auto i = 0u; i < 4u`

File: lib/Arch/X86/Semantics/DATAXFER.cpp [hunk 6]
```diff
@@ -1121,3 +1121,51 @@ IF_64BIT(DEF_ISEL(MOVSXD_GPRv_MEMd_64) = MOVSX<R64W, M32, int64_t>;)
...
+    if (READBIT(k_vec, i) == 0) {
+      dst_vec = SInsertV32(dst_vec, i, 0);
+    }else {
```
Comment (pgoodman) [inline lib/Arch/X86/Semantics/DATAXFER.cpp:1158]: `} else {`

**Review by pgoodman — CHANGES_REQUESTED** [review]
Mostly superficial changes requested. We'd be very interested in those other 296 semantics!

File: lib/Arch/X86/Semantics/DATAXFER.cpp (hunk 1 above)
Comment (yjxxin) [reply lib/Arch/X86/Semantics/DATAXFER.cpp]: Modified according to your opinion

File: lib/Arch/X86/Semantics/DATAXFER.cpp (hunk 4 above)
Comment (yjxxin) [reply lib/Arch/X86/Semantics/DATAXFER.cpp:1145]: Yes, modified according to your opinion

**Comment by pgoodman** [comment]
It looks like this relies on the K registers to be in place, so I'll wait on that PR to be merged. Otherwise, excellent work!

**Review by pgoodman — APPROVED** [review]

**Pull request merged**